Automattic/wp-calypso
# Simplify Happychat language warning logic

#### Changes proposed in this Pull Request

Noticed while working on https://github.com/Automattic/wp-calypso/pull/55498. As Matt explains:

The logic to show the "English only" warning is not correct. If you are a Business/Ecom user but chat is full, it still shows you that support is only in English — even though your ticket is sent to the escalated contractor queue.

This logic should be simplified to show if:

- If you get the Chat form but are in an unsupported locale for Chat
- Or you get the Ticket form but are in an unsupported locale for Tickets

#### Testing instructions

Visit `/help/contact`.

If you're language is English (US) or if you're in a supported locale for the given support form (like `es` for tickets), you should not see the notice:

<img width="740" alt="Screenshot 2021-09-09 at 19 44 00" src="https://user-images.githubusercontent.com/3392497/132752515-2433c4ee-74d8-42dd-a09a-4e286e630b88.png">
<img width="749" alt="Screenshot 2021-09-09 at 19 42 58" src="https://user-images.githubusercontent.com/3392497/132752551-c16cbfa7-6d3f-4969-b24e-3a2d36f013f7.png">

If your locale is not supported (like `pl`), you'll see a notice like this:
<img width="749" alt="Screenshot 2021-09-09 at 19 32 09" src="https://user-images.githubusercontent.com/3392497/132752590-38bff925-29c7-466c-a6f2-140f04881703.png">

There is a known issue related to the fact that `en` != `en-gb`, but that requires more collaboration and planning, so I'll preserve the current handling (meaning English UK is _not_ treated like English US).

Related to https://github.com/Automattic/wp-calypso/pull/55498
